File: manning/Chapter9/CommandProcessor.java
package Chapter9;
import java.io.*;
import java.util.regex.*;
import java.util.*;

public class CommandProcessor
{
  public CommandProcessor() {
  }

  public void processCommands( InputStream in ) throws IOException {
    // BufferedReader lets us read line-by-line
    Reader r = new InputStreamReader( System.in );
    BufferedReader br = new BufferedReader( r );

    Pattern pattern = Pattern.compile( "\\s+" );

    while (true) {
      String line = br.readLine();
      if (line==null)
        break;
      //use space to split words
      String words[] = pattern.split( line );

      if (words[0].equals( "moveto" )) {
        int x = Integer.parseInt( words[1] );
        int y = Integer.parseInt( words[2] );
        moveTo( x, y );
      } else if (words[0].equals( "setname" )) {
        String name = words[1];
        setName( name );
      } else if (words[0].equals( "setbounds" )) {
        int x = Integer.parseInt( words[1] );
        int y = Integer.parseInt( words[2] );
        int w = Integer.parseInt( words[3] );
        int h = Integer.parseInt( words[4] );
        setBounds( x, y, w, h );
      } else {
        System.out.println( "Error: "+line );
      }
    }
  }

  public void moveTo( int x, int y ) {
    System.out.println( "- moveTo "+x+" "+y );
  }

  public void setName( String name ) {
    System.out.println( "- setName "+name );
  }

  public void setBounds( int x, int y, int w, int h ) {
    System.out.println( "- setBounds "+x+" "+y+" "+w+" "+h );
  }

  static public void main( String args[] ) throws IOException {
    CommandProcessor cp = new CommandProcessor();
    cp.processCommands( System.in );
  }
}

File: manning/Chapter9/commandprocessor.pl
#!perl -w

my @vec = ();

while( <> ) {
  chop;

  @words = split( /\s+/ );

  if ($words[0] eq "moveto") {
    # Grab the arguments from the word list
    my $x = $words[1];
    my $y = $words[2];

    # Call the routine that implements the command
    &moveTo( $x, $y );
  } elsif ($words[0] eq "setname") {
    # Grab the arguments from the word list
    my $name = $words[1];

    # Call the routine that implements the command
    &setName( $name );
  } elsif ($words[0] eq "setbounds") {
    # Grab the arguments from the word list
    my $x = $words[1];
    my $y = $words[2];
    my $w = $words[3];
    my $h = $words[4];

    # Call the routine that implements the command
    &setBounds( $x, $y, $w, $h );
  } else {
    # Any unknown command, and we quit
    die "Error: $_";
  }
}

sub moveTo {
  my ($x, $y) = @_;
  print "- moveTo $x $y\n";

}

sub setName {
  my ($name) = @_;
  print "- setName $name\n";
}

sub setBounds {
  my ($x, $y, $w, $h) = @_;
  print "- setBounds $x $y $w $h\n";
}
